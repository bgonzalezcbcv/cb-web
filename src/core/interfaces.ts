import { NavigateOptions } from "react-router-dom";

export interface VisualComponent {
	width?: string;
	height?: string;
}

export interface Teacher {
	ci: string;
	firstName: string;
	lastName: string;
	subjects: string[];
}

export enum UserRoleName {
	Administrador = "Administrador",
	Administrativo = "Administrativo",
	Adscripto = "Adscripto",
	Director = "Director",
	Docente = "Docente",
	Recepcion = "Recepci√≥n",
}

export enum UserRoleColor {
	Administrador = "red",
	Administrativo = "blue",
	Adscripto = "cyan",
	Director = "green",
	Docente = "yellow",
	Recepcion = "black",
}

export enum UserRole {
	Administrador,
	Administrativo,
	Adscripto,
	Director,
	Docente,
	Recepcion,
}

export interface User {
	email: string;
	token: string;
	displayName: string;
	role: UserRole;
}

export interface SidebarItem {
	title: string;
	navigationRoute: string;
	navigationParams?: NavigateOptions;
}

export interface SidebarSection {
	sectionTitle: string;
	items: SidebarItem[];
}

export interface Question {
	id: string;
	question: string;
	answer: string;
}

export interface QuestionCategories {
	category: string;
	questions: Question[];
}

export interface Student {
	question_categories: QuestionCategories[];
}
