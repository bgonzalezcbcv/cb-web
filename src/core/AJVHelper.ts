import _ from "lodash";
import Ajv, { ErrorObject } from "ajv";
import ajvErrors from "ajv-errors";
import ajvFormat from "ajv-formats";

const newAjv = new Ajv({ allErrors: true, verbose: true, strictRequired: true });

export const ajv = ajvErrors(ajvFormat(newAjv));

export function getAjvErrors(ajvValidator = ajv): ErrorObject[] | null | undefined {
	console.log(ajv.errors);
	return ajvValidator.errors;
}

// Used to fill the ErrorList component
export function getParsedErrors(ajvValidator = ajv): Record<string, unknown> {
	if (ajvValidator.errors?.length === 0) return {};
	let requiredId = 0;

	const parsedErrors = ajvValidator.errors?.map((error) => {
		if (error.keyword === "required") {
			requiredId++;
			const missingPropertyName = error.params.missingProperty;
			const title = error.parentSchema?.properties[missingPropertyName].title;

			return {
				id:
					error.instancePath
						.replace("/", "")
						.replaceAll(/\/(\d*)\//gm, "[$1]/")
						.replaceAll("/", ".")
						.replaceAll(/\.(\d*)\./gm, "[$1].") +
					(error.instancePath === "" ? `` : ".") +
					`required${requiredId}`,
				title,
				message: "Campo requerido.",
			};
		}

		return {
			id: error.instancePath
				.replace("/", "")
				.replaceAll(/\/(\d*)\//gm, "[$1]/")
				.replaceAll("/", "."),
			title: error.parentSchema?.title,
			message: error.message,
		};
	});

	console.log(parsedErrors);

	let errorObject = {};

	parsedErrors?.forEach((error) => {
		const { id, title, message } = error;

		errorObject = _.set(errorObject, id, {
			title,
			message,
		});
	});

	return errorObject;
}
