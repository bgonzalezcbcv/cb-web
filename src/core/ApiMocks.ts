import { UserRole } from "./interfaces";
import { DocumentType, UserInfo } from "./Models";

export const mockUser: UserInfo = {
	id: 1,
	role: UserRole.Administrador,
	email: "test@test.com",
	name: "Testing",
	surname: "Tester",
	address: "Avenida Siempre viva 123",
	birthdate: "01-01-1999",
	ci: "11113334",
	phone: "22223333",
	token: "",
	complementary_info: {
		beginning_date: "01-03-1999",
		academic_training: [{ title: "Profesorado de Ingles", date: "01-01-1999", attachment: "" }],
	},
	absences: [
		{
			starting_date: "01-01-2022",
			ending_date: "05-01-2022",
			reason: "Covid",
			attachment: "",
		},
	],
	documents: [
		{
			type: DocumentType.Evaluation,
			attachment: "",
			upload_date: "01-05-2022",
		},
	],
};

export const teachersMock = new Array<UserInfo>(10) //
	.fill(mockUser)
	.map(
		(user, id): UserInfo => ({
			...user,
			id: id,
			groups: [
				{ id: "1", name: "3A", year: "2022" },
				{ id: "2", name: "4B", year: "2022" },
			],
		})
	);
