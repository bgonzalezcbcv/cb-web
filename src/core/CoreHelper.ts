import * as XLSX from "xlsx";

/**
 * @param xlsxFile Excel file with .xlsx format.
 * @returns Array of rows from the excel with the typing desired.
 */
export async function processXLSXtoJSON<Type>(xlsxFile: File): Promise<Type[]> {
	const data = await xlsxFile.arrayBuffer();

	const workbook = XLSX.read(data);

	console.log(XLSX.utils.sheet_to_json<Type>(workbook.Sheets[workbook.SheetNames[0]]));

	return XLSX.utils.sheet_to_json<Type>(workbook.Sheets[workbook.SheetNames[0]]);
}

export function downloadFile(fileName: string, file: Blob): void {
	const url = window.URL.createObjectURL(new Blob([file]));
	const link = document.createElement("a");
	link.href = url;
	link.setAttribute("download", fileName);

	// Append to html link element page
	document.body.appendChild(link);

	// Start download
	link.click();

	// Clean up and remove the link
	link.parentNode?.removeChild(link);
}
