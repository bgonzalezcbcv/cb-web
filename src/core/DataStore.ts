import _ from "lodash";
import { makeAutoObservable, action, reaction } from "mobx";

import { User, Teacher, UserRole } from "./interfaces";

export class DataStore {
	private static instance: DataStore;

	public loggedUser: User | null = null;

	public teachers: Teacher[] = [];

	private constructor() {
		const savedStateJson = localStorage.getItem("store");

		if (savedStateJson) {
			try {
				const savedState = JSON.parse(savedStateJson);

				const clonedState = _.cloneDeep(this);

				Object.keys(this).forEach((attribute) => (clonedState[attribute as keyof DataStore] = savedState[attribute]));

				Object.keys(this).forEach(
					(attribute) =>
						// eslint-disable-next-line
						// @ts-ignore
						(this[attribute as keyof DataStore] = clonedState[attribute])
				);
			} catch (err) {
				console.error(err);
			}
		}

		makeAutoObservable(this);
	}

	public static getInstance(): DataStore {
		if (!DataStore.instance) DataStore.instance = new DataStore();

		return this.instance;
	}

	// todo: need to implement this function.
	@action
	public logIn(): boolean {
		this.loggedUser = {
			email: "testingEmail@xmail.test",
			token: "notAToken",
			displayName: "Juan Prueba",
			role: UserRole.Administrativo,
		};

		return true;
	}

	@action
	public logOut(): void {
		this.loggedUser = null;
	}

	@action
	public addTeacher(teacher: Teacher): void {
		this.teachers = [...this.teachers, teacher];
	}

	@action
	public deleteTeacherByCi(ci: string): void {
		this.teachers = this.teachers.filter((teacher) => teacher.ci !== ci);
	}
}

reaction(
	() => JSON.stringify(DataStore.getInstance()),
	(json) => {
		localStorage.setItem("store", json);
	},
	{
		delay: 500,
	}
);
