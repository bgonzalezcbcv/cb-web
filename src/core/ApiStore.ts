import axios, { Axios } from "axios";

import { Student } from "./Models";
import { User, UserRole } from "./interfaces";

import { DataStore } from "./DataStore";

const dataStore = DataStore.getInstance();

const baseConfig = {
	headers: {
		Authorization: `Bearer ${dataStore.loggedUser?.token}`,
		"Content-Type": "application/json",
	},
	baseURL: process.env["REACT_APP_API_URL"],
};

// todo: use this instance in order to do a middleware.
//eslint-disable-next-line
const axiosInstance = new Axios(baseConfig);

// todo: create middleware in order to log out on 403.

interface SignInResponseData {
	id: number;
	email: string;
	ci: string;
	name: string;
	surname: string;
	birthdate: string;
	address: string;
	role?: UserRole;
}
//todo: make the logout.
export async function login(email: string, password: string): Promise<{ success: boolean; data?: User; err: string }> {
	try {
		const errObject = {
			success: false,
			err: "Error durante el inicio de sesi칩n, volver a intentarlo.",
		};

		const config = {
			...baseConfig,
			method: "post",
			url: "/api/sign_in/",
			headers: {
				"Content-Type": "application/json",
			},
			data: JSON.stringify({
				email,
				password,
			}),
		};

		const response = await axios(config);

		if (!(response.data && response.headers)) return errObject;

		const { name, surname, role }: SignInResponseData = response.data;

		const [bearer, token] = (response.headers["authorization"] ?? "").split(" ");

		if (bearer !== "Bearer") return errObject;

		return { success: true, data: { email, token, name, surname, role: role ?? UserRole.Administrador }, err: "" };
		// eslint-disable-next-line
	} catch (error: any) {
		let err = "";
		if (!error?.response) {
			err = "El servidor no responde";
		} else if (error.response?.status === 400) {
			err = "Usuario o contrase침a inv치lidos";
		} else if (error.response?.status === 401) {
			err = "No autorizado";
		} else {
			err = "Fall칩 el inicio";
		}
		return { success: false, err };
	}
}

export async function createStudent(studentToCreate: Student): Promise<boolean> {
	try {
		const config = {
			...baseConfig,
			method: "post",
			url: "/api/students/",
			data: JSON.stringify({
				student: studentToCreate,
			}),
		};

		const response = await axios(config);

		return response.status === 201;
	} catch (e) {
		return false;
	}
}

// // eslint-disable-next-line
// export async function login(user: string, pass: string): Promise<{ success: boolean; data?: User; err: string }> {
// 	return { success: true, data: { email: "pepe@gmail.com", name: "Pepe Bolsilludo", role: UserRole.Administrador, token: "Bolsilludo" }, err: "" };
// }
