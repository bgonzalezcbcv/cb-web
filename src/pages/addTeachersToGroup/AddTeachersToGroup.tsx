/*eslint-disable*/
import React, { useCallback, useState } from "react";

import { DataGrid, GridApi, GridCellValue, GridColDef } from "@mui/x-data-grid";
import * as APIStore from "../../core/ApiStore";
import { Alert, Box, Button, Card, CircularProgress, Input, Paper, Typography } from "@mui/material";
import useFetchFromAPI, { FetchStatus } from "../../hooks/useFetchFromAPI";
import { Group, UserInfo } from "../../core/Models";

interface TeachersProps {
	rows?: UserInfo[];
	groupName?: string; //por ahora, luego se cambia el tipo
}

export default function AddTeacher(props: TeachersProps) {
	const { rows, groupName } = props;

	const [teachers, setTeachers] = useState<UserInfo[]>(rows ?? []);
	const [searchText, setSearchText] = React.useState("");

	const { fetchStatus, refetch } = useFetchFromAPI(() => APIStore.fetchTeachers(undefined), setTeachers, !rows);

	const onClickAdd = (params: any) => {
		const api: GridApi = params.api;
		const thisRow: Record<string, GridCellValue> = {};

		api.getAllColumns()
			.filter((c: any) => c.field !== "__check__" && !!c)
			.forEach((c: any) => (thisRow[c.field] = params.getValue(params.id, c.field)));
		//endpoint
		alert("No implementado!");
	};

	const onClickRemove = (params: any) => {
		const api: GridApi = params.api;
		const thisRow: Record<string, GridCellValue> = {};

		api.getAllColumns()
			.filter((c: any) => c.field !== "__check__" && !!c)
			.forEach((c: any) => (thisRow[c.field] = params.getValue(params.id, c.field)));
		//endpoint
		alert("No implementado!");
	};

	const columns: GridColDef[] = [
		{ field: "ci", headerName: "CI", disableColumnMenu: false, flex: 2 },
		{ field: "name", headerName: "Nombre", disableColumnMenu: false, flex: 3 },
		{ field: "surname", headerName: "Apellido", disableColumnMenu: false, flex: 3 },
		{
			field: "",
			headerName: "Añadir/Quitar",
			align: "center",
			disableColumnMenu: true,
			sortable: false,
			flex: 1,
			renderCell: (params) => {
				return !((params.row.groups as Group[]) ?? []).find((group): boolean => group.name === groupName) ? (
					<Button
						variant="contained"
						color="success"
						style={{ maxWidth: "10px", maxHeight: "20px", minWidth: "10px", minHeight: "20px" }}
						onClick={() => onClickAdd(params)}>
						+
					</Button>
				) : (
					<Button
						variant="contained"
						color="error"
						style={{ maxWidth: "10px", maxHeight: "20px", minWidth: "10px", minHeight: "20px" }}
						onClick={() => onClickRemove(params)}>
						-
					</Button>
				);
			},
		},
	];

	const toNoFormatText = (str: string) => {
		return str
			.normalize("NFD")
			.replace(/[\u0300-\u036f]/g, "")
			.replace(/ /g, "")
			.toLowerCase();
	};

	const printTable = useCallback((): JSX.Element | null => {
		switch (fetchStatus) {
			case FetchStatus.Fetching:
				return (
					<Box style={{ padding: "20px", display: "flex", justifyContent: "center", alignItems: "center" }}>
						<CircularProgress />
					</Box>
				);
			case FetchStatus.Error:
				return (
					<Box style={{ padding: "20px", display: "flex", justifyContent: "center", alignItems: "center" }}>
						<Alert severity="error" variant="outlined" onClick={refetch}>
							Falló la carga de docentes. Clickear aquí para reintentar.
						</Alert>
					</Box>
				);
			case FetchStatus.Initial:
				const foundItems = teachers.filter((teacher) =>
					Object.values(teacher).some((value) => value && toNoFormatText(value.toString()).includes(toNoFormatText(searchText)))
				);

				return (
					<DataGrid
						style={{ height: 380, width: "100%" }}
						rows={foundItems}
						getRowId={(row) => row.id}
						columns={columns}
						pageSize={5}
						rowsPerPageOptions={[5]}
					/>
				);
			default:
				return null;
		}
	}, [fetchStatus, teachers, searchText]);

	return (
		<Card
			sx={{
				width: "90%",
				padding: "20px",
				alignSelf: "flex-start",
				marginTop: "20px",
				display: "flex",
				flexDirection: "column",
				boxShadow: "rgba(0, 0, 0, 0.25) 0px 3px 8px",
			}}>
			<Box display="flex" justifyContent="flex-start" width="100%">
				<Typography variant="h4"> Grupo:{groupName}</Typography>
				<br></br>
			</Box>
			<Box
				sx={{
					display: "flex",
					justifyContent: "space-between",
					flexDirection: "row",
					marginBottom: "20px",
					padding: "10px",
				}}>
				<Input
					style={{ flex: 3, marginRight: "10%" }}
					id="search"
					type="text"
					placeholder="Buscar..."
					value={searchText}
					onChange={(e: React.ChangeEvent<any>) => setSearchText(e.target.value)}
				/>
			</Box>
			<Paper>{printTable()}</Paper>
		</Card>
	);
}
