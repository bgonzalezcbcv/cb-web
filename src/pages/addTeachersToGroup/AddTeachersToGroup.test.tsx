import React from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { render } from "@testing-library/react";

import * as API from "../../core/ApiStore";
import { teachersMock } from "../../core/ApiMocks";
import AddTeachersToGroup from "./AddTeachersToGroup";

describe("Teachers", () => {
	const groupValue = "3A";

	it("should render the list", async () => {
		jest.spyOn(API, "fetchTeachers").mockResolvedValue({
			success: true,
			data: teachersMock,
			error: "",
		});

		const wrapper = render(
			<MemoryRouter initialEntries={["/addTeachers/1"]}>
				<Routes>
					<Route path="/addTeachers/:id" element={<AddTeachersToGroup groupName={groupValue} />} />
				</Routes>
			</MemoryRouter>
		);

		expect(await wrapper.findByText(`Grupo:${groupValue}`)).toBeVisible();
		expect(await wrapper.findByPlaceholderText(`Buscar...`)).toBeVisible();
		expect(wrapper).toMatchSnapshot(); //TODO que ande
	});

	it("should render an alert on fail", async () => {
		jest.spyOn(API, "fetchTeachers").mockResolvedValue({
			success: false,
			data: teachersMock,
			error: "",
		});

		const wrapper = render(
			<MemoryRouter initialEntries={["/addTeachers/1"]}>
				<Routes>
					<Route path="/addTeachers/:id" element={<AddTeachersToGroup groupName={groupValue} />} />
				</Routes>
			</MemoryRouter>
		);

		expect(await wrapper.findByText(`Falló la carga de docentes. Clickear aquí para reintentar.`)).toBeVisible();
		expect(wrapper).toMatchSnapshot();
	});
});
