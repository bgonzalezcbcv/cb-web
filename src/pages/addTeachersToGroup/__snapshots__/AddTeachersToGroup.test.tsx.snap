// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teachers should render an alert on fail 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j4i0o4-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiBox-root css-1b1a8p0"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
             Grupo:
            3A
          </h4>
          <br />
        </div>
        <div
          class="MuiBox-root css-1tqz4rm"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
            style="flex: 3; margin-right: 10%;"
          >
            <input
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="search"
              placeholder="Buscar..."
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
            style="padding: 20px; display: flex; justify-content: center; align-items: center;"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedError MuiAlert-outlined css-c80q09-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
              >
                Falló la carga de docentes. Clickear aquí para reintentar.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j4i0o4-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-1b1a8p0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
           Grupo:
          3A
        </h4>
        <br />
      </div>
      <div
        class="MuiBox-root css-1tqz4rm"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
          style="flex: 3; margin-right: 10%;"
        >
          <input
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="search"
            placeholder="Buscar..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-0"
          style="padding: 20px; display: flex; justify-content: center; align-items: center;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedError MuiAlert-outlined css-c80q09-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              Falló la carga de docentes. Clickear aquí para reintentar.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Teachers should render the list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j4i0o4-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiBox-root css-1b1a8p0"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
             Grupo:
            3A
          </h4>
          <br />
        </div>
        <div
          class="MuiBox-root css-1tqz4rm"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
            style="flex: 3; margin-right: 10%;"
          >
            <input
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="search"
              placeholder="Buscar..."
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
            style="padding: 20px; display: flex; justify-content: center; align-items: center;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j4i0o4-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-1b1a8p0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
           Grupo:
          3A
        </h4>
        <br />
      </div>
      <div
        class="MuiBox-root css-1tqz4rm"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
          style="flex: 3; margin-right: 10%;"
        >
          <input
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="search"
            placeholder="Buscar..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-0"
          style="padding: 20px; display: flex; justify-content: center; align-items: center;"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
